urllib3/connectionpool.py

```diff
@@ -628,6 +627,12 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
             err = e
 
         finally:
+            if not clean_exit:
+                # We hit some kind of exception, handled or otherwise. We need
+                # to throw the connection away unless explicitly told not to.
+                conn = conn and conn.close()
```

Comment:
I take it that `conn.close()` will return `None`, right? In other words, the result of this will be a state where the locals has `conn = None` and then `self._put_conn` will add `None` back to the pool?

Then we'll go onto the retry logic around L646?